src/main/scala/com/evvo/agent/func.scala

```diff
@@ -23,24 +23,83 @@ abstract class CreatorFunction[Sol](name: String) extends NamedFunction(name) {
 }
 
 /**
-  * A function that produces a new set of solutions based on a set of solutions.
+  * A function that derives a new set of solutions from some input set of solutions.
   *
   * @param name                      This function's name
   * @param numInputs                 The number of solutions to request in the contents of each
   *                                  input set
   * @param shouldRunWithPartialInput Whether this function should run if the number of solutions
   *                                  in the input is less than `numInputs`
   */
-abstract class MutatorFunction[Sol](name: String,
-                                    val numInputs: Int = 32,
-                                    val shouldRunWithPartialInput: Boolean = true)
+abstract class ModifierFunction[Sol](name: String,
+                                     val numInputs: Int = 32,
+                                     val shouldRunWithPartialInput: Boolean = true)
   extends NamedFunction(name) {
   /**
     * @return The function to call to produce new solutions
     */
-  def mutate(sols: IndexedSeq[Scored[Sol]]): TraversableOnce[Sol]
+  def modify(sols: IndexedSeq[Scored[Sol]]): TraversableOnce[Sol]
+}
+
+/**
+  * A function that uses a one-to-one mapping to derive a new set of solutions from some
+  * input set of solutions.
+  */
+abstract class MutatorFunction[Sol](name: String,
+                                    numInputs: Int = 32)
+  extends ModifierFunction[Sol](name, numInputs, shouldRunWithPartialInput = true) {
+  // Pass shouldRunWithPartialInput as true always - we know there is no requirement for more
+  // than 1 solution at a time, so safe to run on any number of solutions.
+
+  override final def modify(sols: IndexedSeq[Scored[Sol]]): TraversableOnce[Sol] = {
+    // We can define modify in terms of mutation, and ask implementors to just provide a one-to-one
+    // mutator function to map over the solutions.
+    sols.map(_.solution).map(mutate)
+  }
+
+  /**
+    * The method to apply to create new solutions.
+    *
+    * @param sol The solution to base the new solution on.
+    * @return The new solution, to be added to the population.
+    */
+  protected def mutate(sol: Sol): Sol
```

Comment:
I think this should be consistent with `ModifierFunction` and take a `Scored[Sol]`, and just change modify to `    sols.map(mutate)`

Reply:
I disagree - if you are just modifying one solution, there's no need to have the score. There are two reasons we used to include the score. One is that the scores are included in the data in the population, so we figured there was no harm to it. The second is that, with multiple solutions, you can use the scores to figure out how to combine the two. 

If you only have one solution, what are you going to do with the score? I can't see this being the most useful by default. Note that none of our mutators use the score of the solutions. If a user really wants to do something with scores, they can extend modifier agent. 

Same goes for crossover agent.

Reply:
I looked it over again and agree with you. It would be trivial for users to extend if they want to use scores anyway